transaction-service/src/main/kotlin/ua/com/foxminded/restClient/kafka/KafkaConsumerService.kt

```diff
@@ -0,0 +1,76 @@
+package ua.com.foxminded.restClient.kafka
+
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.util.Currency
+import java.util.UUID
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.boot.CommandLineRunner
+import org.springframework.data.domain.Pageable
+import org.springframework.kafka.core.reactive.ReactiveKafkaConsumerTemplate
+import org.springframework.stereotype.Service
+import proto.ProtoMessage
+import reactor.core.publisher.Flux
+import ua.com.foxminded.restClient.mapper.ProtoMapper
+import ua.com.foxminded.restClient.service.CurrencyExchangeService
+import ua.com.foxminded.restClient.service.TransactionService
+
+
+@Service
+class KafkaConsumerService constructor(
```

Comment:
what if you have a second topic to consume from? how will your solution for caka extend?

Reply:
If i should use this service i will add topic in config. 
https://github.com/afanasievM/rest-project/blob/kafka/transaction-service/src/main/kotlin/ua/com/foxminded/restClient/config/KafkaConfig.kt

line 23:        return basicReceiverOptions.subscription(Collections.singletonList(topic))
I can add topic here. 
Or If I nedd to use another consumer service i will create new reactiveKafkaConsumerTemplate and inject it to new service